webpack/assets/javascripts/react_app/components/BookmarkForm/BookmarkForm.js

```diff
@@ -21,56 +28,182 @@ const BookmarkForm = ({
   const existsNamesRegex = new RegExp(
     `^(?!(${bookmarks.map(({ name }) => name).join('|')})$).+`
   );
-  const bookmarkFormSchema = Yup.object().shape({
-    name: Yup.string()
-      .max(...maxLengthMsg(254))
-      .required(requiredMsg())
-      .matches(existsNamesRegex, {
-        excludeEmptyString: true,
-        message: __('name already exists'),
-      }),
-    query: Yup.string()
-      .max(...maxLengthMsg(4096))
-      .required(requiredMsg()),
-  });
 
-  const handleSubmit = (values, actions) =>
+  const QUERY_MAX_LENGTH = 4096;
+  const NAME_MAX_LENGTH = 254;
+  const SUCCESS_TYPE = 'success';
+  const ERROR_TYPE = 'error';
+
+  const nameTooLongMsg = sprintf(
+    __('name is too long, max %s'),
+    NAME_MAX_LENGTH
+  );
+  const nameExists = __('name already exists');
+  const queryTooLongMsg = sprintf(
+    __('Query too long, max %s'),
+    QUERY_MAX_LENGTH
+  );
+  const blankInputErrMsg = __("Can't be blank");
+
+  const [name, setName] = useState('');
+  const [query, setQuery] = useState(searchQuery);
+  const [isPublic, setIsPublic] = useState(true);
+
+  const [nameErrorMsg, setNameErrorMsg] = useState('');
+  const [nameValidated, setNameValidated] = useState('');
+  const [queryErrorMsg, setQueryErrorMsg] = useState('');
+  const [queryValidated, setQueryValidated] = useState('');
+
+  const validateName = value => {
+    if (value.length > NAME_MAX_LENGTH) {
+      setNameValidated(ERROR_TYPE);
+      setNameErrorMsg(nameTooLongMsg);
+    } else if (!existsNamesRegex.test(value)) {
+      setNameValidated(ERROR_TYPE);
+      setNameErrorMsg(nameExists);
+    } else {
+      setNameValidated(SUCCESS_TYPE);
+    }
+  };
+
+  const validateQueryLength = value => {
+    if (value.length > QUERY_MAX_LENGTH) {
+      setQueryValidated(ERROR_TYPE);
+      setQueryErrorMsg(queryTooLongMsg);
+    } else if (value.length < 1) {
+      setQueryValidated(ERROR_TYPE);
+      setQueryErrorMsg(blankInputErrMsg);
+    } else {
+      setQueryValidated(SUCCESS_TYPE);
+    }
+  };
+
+  const handleSubmit = () => {
     dispatch(
-      submitForm({
+      APIActions.post({
+        key: 'BOOKMARKS_FORM_SUBMITTED',
         url,
-        values: { ...values, controller },
-        item: 'Bookmarks',
-        message: __('Bookmark was successfully created.'),
-        successCallback: setModalClosed,
-        actions,
+        params: {
+          public: isPublic,
+          controller,
+          query,
+          name,
+        },
+        handleSuccess: setModalClosed,
+        successToast: () => __('Bookmark was successfully created.'),
```

Comment:
I think in other places we've dropped `successfully` because it's redundant. It's already a success toast. In many places we haven't, but we should. As we're updating pages, let's start doing so.
```suggestion
        successToast: () => __('Bookmark was created.'),
```

Though I'd probably trim it down further to:
```suggestion
        successToast: () => __('Bookmark created.'),
```

Reply:
Yep, sure.